lib/src/test/java/space/taran/arklib/data/index/IndexAggregationTest.kt

```diff
@@ -0,0 +1,171 @@
+        shards = createShards(3)
+        indexAggregation = IndexAggregation(shards)
+    }
+
+    @After
+    fun afterEach() {
+        unmockkAll()
+    }
+
+    private fun createShards(amount: Int): Set<RootIndex> {
+        val shards = HashSet<RootIndex>(amount)
+        for (i in 1..amount) {
+            val rootIndex = createRootIndex()
+            shards.add(rootIndex)
+        }
+        return shards
+    }
+
+    private fun createRootIndex(): RootIndex {
+        val rootIndex = mockk<RootIndex>()
+        val resources = createRandomResources(3)
+        val resourceIdSlot = CapturingSlot<ResourceId>()
+        val resourcePaths = createResourceRandomPaths(resources)
+        every { rootIndex.allPaths() } returns resourcePaths
+        every { rootIndex.allIds() } returns resources.map { it.id }.toSet()
+        every { rootIndex.updates } returns MutableSharedFlow<ResourceUpdates>()
+        every { rootIndex.allResources() } returns resources.associateBy({ it.id }, { it })
+        every { rootIndex.getResource(capture(resourceIdSlot)) } answers {
+            val resourceId = resourceIdSlot.captured
+            resources.firstOrNull { resource -> resource.id == resourceId }
+        }
+        every { rootIndex.getPath(capture(resourceIdSlot)) } answers {
+            val resourceId = resourceIdSlot.captured
+            resourcePaths[resourceId]
+        }
+        coEvery { rootIndex.updateAll() } answers {
+            updatedShards += 1
+        }
+        return rootIndex;
+    }
+
+    private fun createRandomResources(amount: Int): Set<Resource> {
+        return (1..amount).map {
+            val resourceId = createRandomResourceId()
+            Resource(resourceId, UUID.randomUUID().toString(), "txt", FileTime.from(Instant.now()))
+        }.toSet()
+    }
+
+    private fun createRandomResourceId(): ResourceId {
+        val crc32 = generateRandomLong(18)
+        val size = Random.nextLong(0, 16364)
+        return ResourceId(size, crc32)
+    }
+
+    private fun generateRandomLong(length: Int): Long {
+        require(length > 0) { "Length must be greater than 0" }
+        val stringBuilder = StringBuilder(length)
+        repeat(length) {
+            val digit = Random.nextInt(1, 10)
+            stringBuilder.append(digit.toStr())
+        }
+        return stringBuilder.toString().toLong()
+    }
+
+    private fun createResourceRandomPaths(resources: Set<Resource>): Map<ResourceId, Path> {
+        return resources.map { resource ->
+            val path = Path(UUID.randomUUID().toString())
+            resource.id to path
+        }.associateBy({ it.first }, { it.second })
+    }
+
+    @Test
+    fun testAllResources() = runBlocking {
+        val expected = shards.map { it.allResources() }.toList().reduce { acc, map -> acc + map }
+        val allResources = indexAggregation.allResources()
+        assertEquals(9, allResources.size) // 3 resources per shard, so 3 shards * 3 resources
```

Comment:
We should also test situation when resource sets from different shards intersect.

E.g.
shard1: a, b, c
shard2: c, d, e
shard3: e, f, g

Aggregation: a, b, c, d, e, f, g